app/controllers/application_controller.rb

```diff
@@ -4,6 +4,7 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception, prepend: true
   before_action :configure_permitted_parameters, if: :devise_controller?
   before_action :persist_last_visited_path
+  before_action :reset_notification_count, if: :current_user
```

Comment:
I created this as I thought the application controller will help with reloading after manually marking as read but didn't work. I left it there though as after page reload it updates.